### PR TITLE
fix error on globalThis type extend

### LINKED ISSUE #30459: Error on globalThis type extension
**TypeScript Version:**  Version 3.4.0-dev.20190316

**Search Terms:**
`globalThis` `Cannot read property 'length' of undefined`

**Code**

```extension.d.ts
declare global {
    namespace globalThis  {
        var test: string;
    }

    namespace foo {
        var bar: string;
    }
}

export {}
```

```index.ts
import "./extention";

globalThis.test = "a-b";
console.log(globalThis.test.split("-"));
```

**Expected behavior:**
compile succeed.

**Actual behavior:**
raise error by tsc

```
/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:72951
                throw e;
                ^

TypeError: Cannot read property 'length' of undefined
    at checkModuleDeclaration (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:47547:44)
    at checkSourceElementWorker (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:48062:28)
    at checkSourceElement (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:47911:17)
    at Object.forEach (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:217:30)
    at checkBlock (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:46001:20)
    at checkSourceElementWorker (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:48017:28)
    at checkSourceElement (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:47911:17)
    at checkModuleDeclaration (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:47594:17)
    at checkSourceElementWorker (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:48062:28)
    at checkSourceElement (/Users/vvakame/.nodebrew/node/v11.10.1/lib/node_modules/typescript/lib/tsc.js:47911:17)
```

**Playground Link:** 
none.

**Related Issues:** 
none.

### PR DESCRIPTION
**Pull request by vvakame**
Fixes #30459

### CODE REVIEW

**Comment by sandersn** [comment]
Thanks for the fast-turnaround fix!

File: tests/cases/compiler/extendGlobalThis.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        var test: string;
+    }
+}
+
+export {}
+
+// @filename: index.ts
+import "./extention";
+
+globalThis.tests = "a-b";
```
Comment (jthemphill) [inline tests/cases/compiler/extendGlobalThis.ts:13]:
```suggestion
globalThis.test = "a-b";
```

File: tests/cases/compiler/extendGlobalThis.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+declare global {
+    namespace globalThis  {
+        var test: string;
+    }
+}
+
+export {}
+
+// @filename: index.ts
+import "./extention";
```
Comment (jthemphill) [inline tests/cases/compiler/extendGlobalThis.ts:11]:
```suggestion
import "./extension";
```

**Review by jthemphill — COMMENTED** [review]
Sorry to randomly drive-by a 5-year old PR... I was looking through test cases to figure out how I should extend `globalThis` in my applications and library, and I found this test case as an example. Thank you for adding it!

I just haven't been able to figure out why we don't expect any typechecker errors in `baselines/reference` for this test case, given that the test case has two typos in it. I don't know if the typos are intentional, but whether intentional or not, the typechecker should report the errors and indeed it does in a real project or [in the Playground](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEBzCB7ARlC8DeBYAUPEfAHZQC2IAzgA5RgLLqYAqAFgJZVF6HH8A3WPAAu1EQC54VETA4lEAbgL8AvgXX4CTDBHZcAdGJncAvPABEUALRoLy-GBQkqKCCAOpEACh2tOVEbiBrQQHCLeFtYWAJQxDgQgAB40KDAiOJpAA).

Is the typechecker running properly in these test files?

**Pull request merged**